Bio/Align/__init__.py

```diff
@@ -926,6 +928,429 @@ def sort(self, key=None, reverse=False):
+        self.score = score
+        self.path = path
+
+    # For Python2 only
+    def __cmp__(self, other):
+        if self.path < other.path:
+            return -1
+        if self.path > other.path:
+            return +1
+        return 0
+
+    def __eq__(self, other):
+        return self.path == other.path
+
+    def __ne__(self, other):
+        return self.path != other.path
+
+    def __lt__(self, other):
+        return self.path < other.path
+
+    def __le__(self, other):
+        return self.path <= other.path
+
+    def __gt__(self, other):
+        return self.path > other.path
+
+    def __ge__(self, other):
+        return self.path >= other.path
+
+    def __format__(self, format_spec):
+        if format_spec == 'psl':
+            return self._format_psl()
+        return str(self)
+
+    def __str__(self):
+        query = self.query
+        target = self.target
+        try:
+            # check if query is a SeqRecord
+            query = query.seq
+        except AttributeError:
+            # query is a Seq object or a plain string
+            pass
+        try:
+            # check if target is a SeqRecord
+            target = target.seq
+        except AttributeError:
+            # target is a Seq object or a plain string
+            pass
+        seq1 = str(target)
+        seq2 = str(query)
+        n1 = len(seq1)
+        n2 = len(seq2)
+        aligned_seq1 = ""
+        aligned_seq2 = ""
+        pattern = ""
+        path = self.path
+        end1, end2 = path[0]
+        if end1 > 0 or end2 > 0:
+            end = max(end1, end2)
+            aligned_seq1 += " " * (end - end1) + seq1[:end1]
+            aligned_seq2 += " " * (end - end2) + seq2[:end2]
+            pattern += ' ' * end
+        start1 = end1
+        start2 = end2
+        for end1, end2 in path[1:]:
+            gap = 0
+            if end1 == start1:
+                gap = end2 - start2
+                aligned_seq1 += '-' * gap
+                aligned_seq2 += seq2[start2:end2]
+                pattern += ' ' * gap
+            elif end2 == start2:
+                gap = end1 - start1
+                aligned_seq1 += seq1[start1:end1]
+                aligned_seq2 += '-' * gap
+                pattern += ' ' * gap
+            else:
+                s1 = seq1[start1:end1]
+                s2 = seq2[start2:end2]
+                aligned_seq1 += s1
+                aligned_seq2 += s2
+                for c1, c2 in zip(s1, s2):
+                    if c1 == c2:
+                        pattern += '|'
+                    else:
+                        pattern += '.'
+            start1 = end1
+            start2 = end2
+        n1 -= end1
+        n2 -= end2
+        n = max(n1, n2)
+        aligned_seq1 += seq1[end1:] + ' ' * (n - n1)
+        aligned_seq2 += seq2[end2:] + ' ' * (n - n2)
+        pattern += ' ' * n
+        text = """\
+%s
+%s
+%s
+""" % (aligned_seq1, pattern, aligned_seq2)
+        return text
```

Comment:
Style wise, I would avoid the multi-line triple-quoted string and we replace this with:

```python
        # Avoiding trailing write space:
        return "%s\n%s\n%s\n" % (aligned_seq1, pattern, aligned_seq2)
```

Furthermore, as per my main comment about the doctests having trailing whitespace, would this be acceptable to you:

```python
        return "%s\n%s\n%s\n" % (aligned_seq1, pattern.rstrip(), aligned_seq2)
```

Reply:
I will use your first suggestion